engine/src/main/java/org/hibernate/search/query/engine/impl/HSQueryImpl.java

```diff
@@ -670,7 +678,32 @@ else if ( this.sort != null && projection != null ) {
 		);
 	}
 
+	private void validateSortFields(ExtendedSearchIntegrator extendedIntegrator) {
+		SortField[] sortFields = sort.getSort();
+		for ( SortField sortField : sortFields ) {
+			if ( sortField instanceof DistanceSortField ) {
+				validateDistanceSortField( extendedIntegrator, sortField );
+			}
+		}
+	}
 
+	private void validateDistanceSortField(ExtendedSearchIntegrator extendedIntegrator, SortField sortField) {
+		String field = sortField.getField();
```

Comment:
I would like to add a test for this scenario. Do I just need to specify the document id as sort field?